vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -940,20 +940,19 @@ public void failure(VimeoError error) {
             parameters.put(Vimeo.PARAMETER_VIDEO_DESCRIPTION, description);
         }
 
-        if (privacyValue != null) {
-            if (privacyValue == Privacy.PrivacyValue.PASSWORD) {
-                if (password == null) {
-                    callback.failure(new VimeoError("Password cannot be null password privacy type"));
+        if (privacySettingsParams != null && !privacySettingsParams.getParams().isEmpty()) {
+            final Map<String, Object> privacyMap = privacySettingsParams.getParams();
+            final PrivacyViewValue viewPrivacyValue = (PrivacyViewValue) privacyMap.get(Vimeo.PARAMETER_VIDEO_VIEW);
```

Comment:
It would be better if this unsafe cast was not necessary. One option could be to add a `.doesRequirePassword(): Boolean` method to `PrivacySettingsParams`. I don't think it is absolutely necessary to change in this PR since we control the creation of the map, maybe we can revisit it later.

Reply:
I will take a note of it and come back to it later.